iocore/net/ALPNSupport.cc

```diff
@@ -98,18 +98,24 @@ ALPNSupport::select_next_protocol(SSL *ssl, const unsigned char **out, unsigned
 {
   const unsigned char *npnptr = nullptr;
   unsigned int npnsize        = 0;
-  if (this->getNPN(&npnptr, &npnsize)) {
+  int retval                  = SSL_TLSEXT_ERR_ALERT_FATAL;
```

Comment:
I don't think we can use this error code for 1.0.2.
https://www.openssl.org/docs/man1.0.2/man3/SSL_set_alpn_protos.html

Reply:
Hmm, how does one express the failure without the SSL_TLSEXT_ERR_ALERT_FATAL?  I'll dig into the 1.0.2 code base and add an ifdef if necessary.

Reply:
Returning `SSL_TLSEXT_ERR_ALERT_FATAL` isn't harmful, but it's impossible to distinguish no-match from no-negotiation.
https://github.com/openssl/openssl/blob/12ad22dd16ffe47f8cde3cddb84a160e8cdb3e30/ssl/t1_lib.c#L2087-L2105

Reply:
Hmm that is unfortunate.  However, I think our time for worrying about 1.0.2 is coming to a close.